x/confidentialtransfers/types/zk.go

```diff
@@ -419,3 +419,95 @@ func (a *Auditor) FromProto() (*TransferAuditor, error) {
 		TransferAmountHiEqualityProof: transferAmountHiEqualityProof,
 	}, nil
 }
+
+func (z *ZeroBalanceProof) Validate() error {
+	if z.YP == nil || z.YD == nil || z.Z == nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "zero proof is invalid")
+	}
+	return nil
+}
+
+func (z *ZeroBalanceProof) ToProto(zkp *zkproofs.ZeroBalanceProof) *ZeroBalanceProof {
```

Comment:
Not actional for this PR but i'm planning to rework all the ToProto calls to make them just regular constructors `NewZeroBalanceProof(zkp*zkproofs.ZeroBalanceProof) *ZeroBalanceProof` instead of a instance method of z, since it doesn't actually use z.